documents_stock_production_lot/models/documents_workflow_rule.py

```diff
@@ -0,0 +1,59 @@
+# Copyright (C) 2021 Open Source Integrators
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+
+class DocumentsWorkflowRule(models.Model):
+    _inherit = ["documents.workflow.rule"]
+
+    def _compute_business(self):
+        return self._get_business()
+
+    has_business_option = fields.Boolean(default=True, compute="_compute_business")
+    create_model = fields.Selection(
+        selection_add=[("stock.production.lot", "Lot/Tracking Number")]
+    )
+
+    def create_record(self, documents=None):
+        res = super().create_record(documents=documents)
+        if self.create_model == "stock.production.lot":
+            lot = self.env.company.lot_id
+            if not lot:
+                raise UserError(
+                    _(
+                        """Missing Document lot number configuration,
+                        please configure on existing lot number under the 
+                        Configration > Settings > Documents"""
+                    )
+                )
+
+            for document in documents:
+                # this_document is the document in use for the workflow
+                this_document = document
+                if (
+                    document.res_model or document.res_id
+                ) and document.res_model != "documents.document":
+                    attachment_copy = document.attachment_id.with_context(
+                        no_document=True
+                    ).copy()
+                    this_document = document.copy({"attachment_id": attachment_copy.id})
+                this_document.write(
+                    {
+                        "res_model": lot._name,
+                        "res_id": lot.id,
+                    }
+                )
```

Comment:
@Vimal-OSI This is linking the documents to the same lot... Not sure this is the expected behavior. We don't want to create a new lot, we want to attach the document to an existing lot.